_posts/2018-03-01-Rust-1.24.1.md

```diff
@@ -0,0 +1,257 @@
+> compilation. The compiler currently has logic for detecting a failure to
+> spawn and instead passing arguments via a file instead, but this failure
+> detection only triggers if a process actually fails to spawn.
+
+However, when generating that file, we were doing it incorrectly. As [the
+docs state](https://docs.microsoft.com/en-gb/cpp/build/reference/unicode-support-in-the-compiler-and-linker#linker-response-files-and-def-files):
+
+> Response files and DEF files can be either UTF-16 with a BOM, or ANSI.
+
+We were providing a UTF-8 encoded file, with no
+[BOM](https://en.wikipedia.org/wiki/Byte_order_mark). The fix is therefore
+straightforward: produce a UTF-16 file with a BOM.
+
+### Make the error index generator work again
+
+When packaging Rust for various Linux distros, it was found that [building
+1.24 with 1.24 fails](https://github.com/rust-lang/rust/issues/48308).
+This was caused by an incorrect path, causing certain metadata to not
+be generated properly.
+
+### Cargo will warn on Windows 7 if an update is needed.
+
+In February of 2017, [GitHub announced that they were dropping support for
+weak cryptographic
+standards](https://githubengineering.com/crypto-deprecation-notice/). One
+year later, in February of 2018, [the deprecation period is over, and support
+is
+removed](https://blog.github.com/2018-02-23-weak-cryptographic-standards-removed/).
+In general, this is a great thing.
+
+Cargo uses GitHub to store the index of Crates.io, our package repository.
+It also uses `libgit2` for `git` operations. `libgit2` uses
+[WinHTTP](https://msdn.microsoft.com/en-us/library/windows/desktop/aa382925(v=vs.85).aspx)
+for making HTTP calls. As part of the OS, its feature set depends on the OS you're using.
+
+> This section uses "Windows 7" to mean "Windows 7, Windows Server 2018, and Windows Server 2012",
+> because it's much shorter. The following applies to all three of these editions of Windows,
+> however.
+
+Windows 7 [received an update](https://support.microsoft.com/en-us/help/3140245/update-to-enable-tls-1-1-and-tls-1-2-as-a-default-secure-protocols-in)
+in June of 2016 regarding TLS. Before the patch, Windows 7 would use TLS 1.0 by default. This update
+allows for applications to use TLS 1.1 or 1.2 natively instead.
+
+If your system has not received that update, then you'd still be using TLS 1.0. This means
+that accessing GitHub would start to fail.
+
+`libgit2` [created a fix](https://github.com/libgit2/libgit2/pull/4550), using the `WinHTTP` API
+to request TLS 1.2. On master, we've [updated to fix this](https://github.com/rust-lang/cargo/pull/5091),
+but for 1.24.1 stable, we're [issuing a warning](https://github.com/rust-lang/cargo/pull/5069),
+suggesting that they upgrade their Windows version. There's a balance here: we could have backported
+the `libgit2` fixes, but that'd require updating a lot of code, which feels incorrect for a point
+release, especially since the issue does not affect patched systems.
```

Comment:
Suggested rewording for last sentence, removing the "balance" phrasing:

> Although the `libgit2` fix could have been backported, we felt that the code change footprint was too large for the point release, especially since..."